src/codecs/Reaction.ts

```diff
@@ -0,0 +1,60 @@
+import { ContentTypeId, ContentCodec, EncodedContent } from '../MessageContent'
+
+// xmtp.org/text
+//
+// This content type is used for a plain text content represented by a simple string
+export const ContentTypeReaction = new ContentTypeId({
+  authorityId: 'xmtp.org',
+  typeId: 'reaction',
+  versionMajor: 1,
+  versionMinor: 0,
+})
+
+export type Reaction = {
+  /**
+   * The message ID for the message that is being reacted to
+   */
+  reference: string
+  /**
+   * The action of the reaction
+   */
+  action: 'added' | 'removed'
+  /**
+   * The content of the reaction
+   */
+  content: string
+}
+
+export type ReactionParameters = Pick<Reaction, 'action' | 'reference'> & {
+  encoding: 'UTF-8'
+}
+
+export class ReactionCodec implements ContentCodec<Reaction> {
+  get contentType(): ContentTypeId {
+    return ContentTypeReaction
+  }
+
+  encode(content: Reaction): EncodedContent<ReactionParameters> {
+    return {
+      type: ContentTypeReaction,
+      parameters: {
+        encoding: 'UTF-8',
+        action: content.action,
+        reference: content.reference,
+      },
+      content: new TextEncoder().encode(content.content),
+    }
+  }
+
+  decode(content: EncodedContent<ReactionParameters>): Reaction {
+    const encoding = content.parameters.encoding
+    if (encoding && encoding !== 'UTF-8') {
+      throw new Error(`unrecognized encoding ${encoding}`)
+    }
+    return {
+      action: content.parameters.action,
+      reference: content.parameters.reference,
+      content: new TextDecoder().decode(content.content),
```

Comment:
Do we want to put some limit (even a relatively large one like 1kb) on the size of the content? I can't think of a valid use-case for large content

Reply:
base64 encoded SVG, images, etc... could be a thing 🤔

Reply:
Hmmm. For something that different from an emoji, we wouldn't want some other metadata to tell the recipient "decode this like an image".

Is that really an intended use-case of this version of the content type?

Reply:
AFAIK for this initial version, we're supporting the standard set of emojis via string tokens, so `smile` => 😄 (`:smile:`). this will allow clients to use their own custom emojis like Slack. i don't think the intent of this content type is to store actual content like images, SVGs, etc. so it makes sense to limit the length.